validator/sawtooth_validator/server/cli.py

```diff
@@ -349,27 +349,27 @@ def main(args=None):
         'Starting validator with %s scheduler',
         validator_config.scheduler)
 
-    validator = Validator(
-        bind_network,
-        bind_component,
-        endpoint,
-        validator_config.peering,
-        validator_config.seeds,
-        validator_config.peers,
-        path_config.data_dir,
-        path_config.config_dir,
-        identity_signer,
-        validator_config.scheduler,
-        validator_config.permissions,
-        validator_config.minimum_peer_connectivity,
-        validator_config.maximum_peer_connectivity,
-        validator_config.network_public_key,
-        validator_config.network_private_key,
-        roles=validator_config.roles,
-        metrics_registry=wrapped_registry)
-
     # pylint: disable=broad-except
     try:
+        validator = Validator(
```

Comment:
To move this within the block, you should set validator = None prior to the block, then make sure the finally clause checks "if validator is not None" before using it.  As it is, a well-timed KeyboardInterrupt may cause an exception in the finally block.